File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        arr = []
        map = {}

        for i in range(len(nums)):
            diff = target - nums[i]

            if (diff in map):
                arr = [i, map[diff]]
            else:
                map[nums[i]] = i
                

        return arr

        

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        int[] ans = new int[2];

        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            if (map.containsKey(diff)) {
                ans[0] = i;
                ans[1] = map.get(diff);
                return ans;           
            }

            map.put(nums[i], i);

        }

        return ans;
        
    }
}